lld/ELF/LinkerScript.cpp

```diff
@@ -1196,9 +1202,17 @@ void LinkerScript::adjustOutputSections() {
     if (sec->sectionIndex != UINT32_MAX)
       maybePropagatePhdrs(*sec, defPhdrs);
 
+    // Discard .relro_padding if we have not seen one non-NOBITS RELRO section.
+    // Note: when .tbss is the only RELRO section, it's difficult to pick a
+    // suitable padding size (see computeFilOffset). For simplicity, don't
+    // retain .relro_padding in this case.
+    if (in.relroPadding && in.relroPadding->getParent() == sec && !seenRelro)
+      discardable = true;
     if (discardable) {
       sec->markDead();
       cmd = nullptr;
+    } else {
+      seenRelro |= sec->relro && sec->type != SHT_NOBITS;
```

Comment:
Would it be worth restricting this to sec->type != SHT_NOBITS && (sec->flags & SHF_TLS)

Reply:
Looks good. Added a new test to `relocation-copy-relro.s` where .bss.rel.ro is the only RELRO section and we still append the padding.